engines/tot/dialog.cpp

```diff
@@ -0,0 +1,676 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common/scummsys.h"
+#include "common/textconsole.h"
+
+#include "tot/dialog.h"
+#include "tot/tot.h"
+
+namespace Tot {
+
+Common::String decrypt(Common::String encryptedText) {
+	for (int i = 0; i < encryptedText.size(); i++) {
+		encryptedText.setChar(g_engine->_decryptionKey[i] ^ (char)encryptedText[i], i);
+	}
+	return encryptedText;
+}
+
+Common::List<uint>* findDialogue(Tree tree, byte characterIndex);
+
+static void findDownwards(Tree curTree, bool &descend) {
+	if (curTree != NULL) {
+		if (curTree->element.spoken != '1') {
+			descend = true;
+			return;
+		} else {
+			findDownwards(leftChild(curTree), descend);
+			findDownwards(rightSibling(curTree), descend);
+		}
+	}
+}
+
+Common::List<uint>* findDialogue(Tree tree, byte characterIndex) {
+	bool speak, ascend, descend, border, forward;
+
+	Tree auxTree = tree->child;
+	bool done = false;
+	Common::List<uint> *linkedList = new Common::List<uint>();
+	border = false;
+	ascend = false;
```

Comment:
Missing initialization of variables, causing a failure in compilation under Windows

```suggestion
	bool speak = false, ascend = false, descend = false, border = false, forward = false, done = false;

	Tree auxTree = tree->child;
	Common::List<uint> *linkedList = new Common::List<uint>();
```